CMakeLists.txt

```
@@ -0,0 +1,338 @@
+# note: previous deterministic and optimisation report options not included
+set (NCI_INTEL_FLAGS "-r8 -i4 -w -fpe0 -ftz -convert big_endian -assume byterecl -check noarg_temp_created")
+set (NCI_REPRO_FLAGS "-fp-model precise -fp-model source -align all")
+set (NCI_OPTIM_FLAGS "-g3 -O2 -axCORE-AVX2 -debug all -check none -traceback -assume buffered_io")
+set (NCI_DEBUG_FLAGS "-g3 -O0 -debug all -check all -no-vec -traceback -assume nobuffered_io")
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-Wall -fdefault-real-8 -fdefault-double-8 -ffpe-trap=invalid,zero,overflow -fallow-argument-mismatch")
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "Intel")
+  set(CMAKE_Fortran_FLAGS_RELEASE "${NCI_INTEL_FLAGS} ${NCI_REPRO_FLAGS} ${NCI_OPTIM_FLAGS}")
+  set(CMAKE_Fortran_FLAGS_DEBUG "${NCI_INTEL_FLAGS} ${NCI_REPRO_FLAGS} ${NCI_DEBUG_FLAGS}")
+else()
+  message(WARNING "Fortran compiler with ID ${CMAKE_Fortran_COMPILER_ID} will be used with CMake default options")
+endif()
+
+add_compile_definitions(
+  NXGLOB=${CICE_NXGLOB}
+  NYGLOB=${CICE_NYGLOB}
+  BLCKX=${CICE_BLCKX}
+  BLCKY=${CICE_BLCKY}
+  MXBLCKS=${CICE_MXBLCKS}
+  NUMIN=${NUMIN}
+  NUMAX=${NUMAX}
+  TRAGE=${TRAGE}
+  TRFY=${TRFY}
+  TRLVL=${TRLVL}
+  TRPND=${TRPND}
+  NTRAERO=${NTRAERO}
+  TRBRI=${TRBRI}
+  NBGCLYR=${NBGCLYR}
+  TRBGCS=${TRBGCS}
+  NICECAT=${NICECAT}
+  NICELYR=${NICELYR}
+  NSNWLYR=${NSNWLYR}
+)
+
+add_compile_definitions(
+  gather_scatter_barrier
+  ncdf
+  OASIS3_MCT
+)
+
+if (CICE_IO MATCHES PIO)
+  add_compile_definitions(PIO)
+endif()
+
+# ???
+# setenv CHAN     MPI1	  # MPI1 or MPI2 (always MPI1!) 
+
+if(CICE_DRIVER MATCHES "auscom")
+  add_compile_definitions(AusCOM coupled)
+elseif(CICE_DRIVER MATCHES "access")
+  add_compile_defitions(ACCESS AusCOM coupled)
+endif()
+
+
+## Fortran modules path; currently this is simply set to be the include dir
+set(CMAKE_INSTALL_MODULEDIR ${CMAKE_INSTALL_INCLUDEDIR}
+  CACHE STRING
+  "Fortran module installation path (Not a cmake native variable)"
+)
+
+#[==============================================================================[
+#                              External packages                                #
+#]==============================================================================]
+
+
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(OASIS3PSMILE REQUIRED IMPORTED_TARGET "oasis3-psmile.MPI1")
+pkg_check_modules(OASIS3MCT REQUIRED IMPORTED_TARGET "oasis3-mct")
+pkg_check_modules(OASIS3MPEU REQUIRED IMPORTED_TARGET "oasis3-mpeu")
+pkg_check_modules(OASIS3SCRIP REQUIRED IMPORTED_TARGET "oasis3-scrip")
+
+find_package(MPI REQUIRED COMPONENTS Fortran)
+if(CICE_DRIVER MATCHES "auscom")
+  find_package(LIBACCESSOM2)
+endif()
+
+if(CICE_IO MATCHES "PIO" AND NOT TARGET PIO::PIO_Fortran)
+  find_package(PIO 2.5.3 REQUIRED COMPONENTS Fortran)
+  # Code has not been tested with versions older than 2.5.3, but probably still works fine
```

Just putting comment before code, as for NetCDF block

```suggestion
  # Code has not been tested with versions older than 2.5.3, but probably still works fine
  find_package(PIO 2.5.3 REQUIRED COMPONENTS Fortran)
```